### PR TITLE
vaadinPrepareFrontend and vaadinBuildFrontend are now automatically a…

### LINKED ISSUE #48: Make vaadinPrepareFrontend and vaadinBuildFrontend activate automatically
Currently the production mode auto-activates when you call Gradle with task `vaadinBuildFrontend`. That's because Gradle doesn't use profiles the way Maven does. However, this approach is pretty fragile:
1. There is a bug: the production mode doesn't auto-activate if you just call `./gradlew` and you have `defaultTasks("clean", "vaadinBuildFrontend", "build")` in your build.gradle file.
2. The production mode doesn't auto-activate if you have `build.dependsOn("vaadinBuildFrontend")` in your build.gradle file.
3. You can enforce the production mode by using `vaadin { productionMode = true }` in your `build.gradle` file, but it's not a solid approach to control this by editing `build.gradle` - you don't want your CI to edit your `build.gradle`

We should either fix this behavior for the productionMode to auto-activate properly whenever `vaadinBuildFrontend` task is to be executed, or we should make the productionMode setting respond to e.g. `./gradlew blah blah -Dvaadin.productionMode=true`.

I wonder if there is any merit in having a WAR file with development mode. The disadvantage is apparent: customers would drop a WAR file into production, and the WAR file will start downloading and populating node_modules. But on the development machine one can wish to e.g. deploy a WAR file into local Tomcat for development purposes, which is also a valid approach. And vice versa - some people who do not want to touch JavaScript at all, will be pretty fine developing having the productionMode turned on.

Sounds to me that the best strategy is to let the developer flip the `productionMode` switch (either by system property, or permanently in build.gradle), and the build should accommodate accordingly (e.g. auto-call `vaadinBuildFrontend` but only when the productionMode is true; and auto-call `vaadinPrepareFrontend` always when the plugin is applied).

This would however auto-create `frontend/` folders for every java-based subproject. We should document cleanly how to use the plugin in multi-project builds:

```
plugins {
  id 'java'
  id "com.vaadin" version "0.5.1" apply false
}

project("lib") {
  apply plugin: 'java'
}
project("web") {
  apply plugin: 'war'
  apply plugin: "com.vaadin"
}
```

### PR DESCRIPTION
**Pull request by mvysny**
…ttached to the build cycle. Fixes #48

### CODE REVIEW

**Comment by mvysny** [comment]
Please do not merge just yet:
1. I'll test this in a larger project first
2. @mstahv would like to verify this approach with the Gradle community

File: README.md [hunk 1]
```diff
@@ -48,14 +48,22 @@ There are the following tasks:
   and will place that by default into the `build/vaadin-generated` folder. The folder is
   then later picked up by `jar` and `war` tasks which then package the folder contents properly
   onto the classpath. Note that this task is not automatically hooked into `war`/`jar`/`assemble`/`build` and
-  need to be invoked explicitly.
+  need to be invoked explicitly. WARNING: this task will be skipped if `productionMode` is set to false.
```
Comment (caalador) [inline README.md]: Should this just be `NOTE:` ?

File: README.md [hunk 2]
```diff
@@ -70,8 +78,10 @@ vaadin {
 
 All configuration options follow. Note that you **RARELY** need to change anything of the below.
 
-* `productionMode = false`: Whether or not we are running in productionMode.
-  The `vaadinBuildFrontend` task will automatically switch this to true, there is no need for you to configure anything.
+* `productionMode = false`: Whether or not the plugin should run in productionMode. Defaults to false.
+  Responds to the `-Pvaadin.productionMode` property. You need to set this to `true` if you wish
+  to build a production-ready war/jar artifact. If this is false, the `vaadinBuildFrontend`
+  task is automatically skipped.
```
Comment (caalador) [inline README.md]: This is sort of false as in principle you could build a development mode war that doesn't start the devServer by running build-frontend with flow-build-info containing the property  `enableDevServer=false` which makes it possible tohave a "development mode" war that can be executed on other servers.

File: src/main/kotlin/com/vaadin/gradle/VaadinBuildFrontendTask.kt [hunk 3]
```diff
@@ -58,6 +58,14 @@ open class VaadinBuildFrontendTask : DefaultTask() {
         project.tasks.withType(Jar::class.java) { task: Jar ->
             task.mustRunAfter("vaadinBuildFrontend")
         }
+
+        setOnlyIf {
+            val productionMode: Boolean = VaadinFlowPluginExtension.get(project).productionMode
+            if (!productionMode) {
+                logger.info("Skipping vaadinBuildFrontend since we're not in production mode")
+            }
+            productionMode
```
Comment (caalador) [inline src/main/kotlin/com/vaadin/gradle/VaadinBuildFrontendTask.kt]: As noted in documentation this should preferably also be enabled if `enableDevServer` is false as that would be the same as executing both goals in maven without productionMode=true.

File: README.md (hunk 2 above)
Comment (mvysny) [reply README.md]: Thanks! In such case the `enableDevServer` configuration should be also exposed in VaadinFlowPluginExtension. But first, can you please explain what is the advantage of having a development mode WAR with webpack bundle? Such an information would be useful to have documented both in README and in the javadoc.

File: README.md (hunk 1 above)
Comment (mvysny) [reply README.md]: Yeah, it may be just NOTE. I'll update this and add the info regarding the `enableDevServer` also.

File: README.md (hunk 2 above)
Comment (caalador) [reply README.md]: The use case is basically being able to get all the development time log and features when testing the war for instance on an external server.

The original issue is probably:
https://github.com/vaadin/flow/issues/5876

Comment (mvysny) [reply README.md]: I can not find the `enableDevServer` configuration property in flow-maven-plugin sources, in the 2.1 branch. Is it something new?

Comment (caalador) [reply README.md]: Maven plugin sets it in `BuildFrontendMojo.updateBuildFile`

File: src/main/kotlin/com/vaadin/gradle/VaadinBuildFrontendTask.kt (hunk 3 above)
Comment (mvysny) [reply src/main/kotlin/com/vaadin/gradle/VaadinBuildFrontendTask.kt]: Ah, the `enableDevServer` from the `flow-build-info.json` file. Understood, should now be fixed.

File: README.md (hunk 2 above)
Comment (mvysny) [reply README.md]: Right, the one from the `flow-build-info.json` file. Thanks, should now be fixed.

**Pull request merged**